dev/src/Command/AddComponentCommand.php

```diff
@@ -201,4 +242,73 @@ private function loadProtoContent(string $proto): string
         $response = $client->get($protoUrl);
         return (string) $response->getBody();
     }
+
+    private function bazelBuildLibrary(string $libraryName, string $googleApisDir): bool
+    {
+        $command = 'which bazel';
+        $output = exec($command, $output, $return);
+        if (strlen($output) == 0) {
+            exit("Error: Bazel is not installed. Please install Bazel before running this script.\n");
+        }
+        chdir($googleApisDir);
+        $command = "bazel query 'filter(\"-(php)$\", kind(\"rule\", //google/cloud/" . $libraryName .
+            "/...:*))' | grep -v -E \":(proto|grpc|gapic)-.*-php$\"";
+        exec($command, $output, $return);
+        if ($return === 0 && !empty($output)) {
+            $command = "bazel build " . $output[0];
+            exec($command, $output, $return);
+            if ($return === 0) {
+                chdir($this->rootPath);
+                return true;
+            }
+        }
+        chdir($this->rootPath);
```

Comment:
using the `Process` component should eliminate the need to use shell commands like `exec` and `chdir`, which we want to avoid.